### PR TITLE
fastai compatibility

### PR DESCRIPTION
**Pull request by mgrankin**
1. Remove `device` from nn.Module (most)
2. Split TabNet in two classes. One for processing embeddings (framework-specific) and one for pure TabNet.

### CODE REVIEW

**Comment by eduardocarvp** [comment]
Thank you for your PR, @mgrankin, we are reviewing it on our side!

As for the checks that have failed, they are all related to linting. You can try `make lint` from the tabnet directory (outside the container, it basically runs `$(DOCKER_RUN) 'poetry run flake8') to see where and why flake8 is failing. Please let us know if you have any questions : )

**Comment by mgrankin** [comment]
Thanks @eduardocarvp, I've fixed the formatting.

File: pytorch_tabnet/tab_network.py [hunk 1]
```diff
@@ -23,37 +23,38 @@ class GBN(torch.nn.Module):
...
             y = self.bn(x_)
             res = torch.cat([res, y], dim=0)
 
         return res
 
 
-class TabNet(torch.nn.Module):
-    def __init__(self, input_dim, output_dim, n_d=8, n_a=8,
-                 n_steps=3, gamma=1.3, cat_idxs=[], cat_dims=[], cat_emb_dim=1,
+class DryTabNet(torch.nn.Module):
```
Comment (Optimox) [inline pytorch_tabnet/tab_network.py]: could you rename `DryTabNet` into `TabNetNoEmbeddings`? I think it's a bit more expressive that way.

File: pytorch_tabnet/tab_network.py [hunk 2]
```diff
@@ -23,37 +23,38 @@ class GBN(torch.nn.Module):
...
+    def __init__(self, input_dim, output_dim,
+                 n_d=8, n_a=8,
+                 n_steps=3, gamma=1.3,
                  n_independent=2, n_shared=2, epsilon=1e-15,
-                 virtual_batch_size=128, momentum=0.02, device_name='auto'):
+                 virtual_batch_size=128, momentum=0.02):
         """
-        Defines TabNet network
+        Defines the essence of TabNet network
```
Comment (Optimox) [inline pytorch_tabnet/tab_network.py]: what about:

`Defines main part of the TabNet network without the embedding layers.`

**Comment by mgrankin** [comment]
Screenshot 2020-02-13 at 18 44 21

File: pytorch_tabnet/tab_network.py (hunk 1 above)
Comment (Optimox) [reply pytorch_tabnet/tab_network.py]: thx!

**Comment by Optimox** [comment]
Thanks @mgrankin, everything looks ok. We'll make a new release today!

**Comment by mgrankin** [comment]
I'm looking forward to seeing the package release. So I can delete the duplicate code in my package.

**Pull request merged**